ui/components/component-library/button-base/button-base.types.ts

```diff
@@ -0,0 +1,112 @@
+import { ReactNode } from 'react';
+import type { BoxProps } from '../../ui/box/box.d';
+import {
+  Color,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { TextDirection, TextProps, ValidTag } from '../text';
+import { Icon, IconName } from '../icon';
+
+export enum ButtonBaseSize {
+  SM = 'sm',
+  MD = 'md',
+  LG = 'lg',
```

Comment:
Our enum conventions should use PascalCase for keys too! Sorry some are a little mixed.
```suggestion
  Sm = 'sm',
  Md = 'md',
  Lg = 'lg',
```